plugin/trino-sqlserver/src/main/java/io/trino/plugin/sqlserver/SqlServerClient.java

```diff
@@ -319,18 +322,28 @@ protected String renameTableSql(String catalogName, String remoteSchemaName, Str
     }
 
     @Override
-    protected String renameColumnSql(JdbcTableHandle handle, JdbcColumnHandle jdbcColumn, String newRemoteColumnName)
+    public void renameColumn(ConnectorSession session, JdbcTableHandle handle, JdbcColumnHandle jdbcColumn, String newColumnName)
     {
         RemoteTableName remoteTableName = handle.asPlainTable().getRemoteTableName();
-        return format(
-                "sp_rename %s, %s, 'COLUMN'",
-                singleQuote(remoteTableName.getCatalogName().orElse(null), remoteTableName.getSchemaName().orElse(null), remoteTableName.getTableName(), "[" + escape(jdbcColumn.getColumnName()) + "]"),
-                "[" + newRemoteColumnName + "]");
+        try (Connection connection = connectionFactory.openConnection(session)) {
+            String columnFrom = DOT_JOINER.join(remoteTableName.getCatalogName().orElse(null), remoteTableName.getSchemaName().orElse(null), remoteTableName.getTableName(), doubleQuote(escape(jdbcColumn.getColumnName())));
```

Comment:
Is the double quote ans escape needed here since we use prepared statement?

Reply:
I just checked all possible variants that sql server can accept and the best way it works when we use quotes around table name instead of brackets and escape quotes inside table name. So I was not able to break it. But if use brackets as quotes in some cases it's not working.

Reply:
it's not connected to prepared statement

Reply:
I thought that since the `columnFrom` is only used as argument to the prepared statement why do we even need to perform escaping or quoting - the server or the driver would do that?

Reply:
As far as I understand, because we use `columnFrom` as argument for prepared statement we don't need quoting anymore. However columnFrom consist of multiple parts like database/schema/table/column `SQLPOOL1.dbo.test2.colum_name` and if you want to have special characters like brackets and double quote inside column name like `[column" name]` you need to quote only column in double quotes and double your double quotes:
`SQLPOOL1.dbo.test2."[colum"" name]"`

Reply:
Isn't it possible then for the catalog or table name to have special characters as well?

Reply:
to be honest didn't check such cases

Reply:
I was able to rename table to contain special brackets:
`alter table test2 rename to "[test2]";`
`show tables;`
`  Table`
`----------`
 `[test2]`

But looks like we don't support to query such tables:
`select * from "[test2]";`
`uery 20220924_171606_00044_winm8 failed: Table 'dbo.[test2]' has no supported columns (all 0 columns are not supported)`

Reply:
But still I was I able to rename column in such specific table using the same rule of double quoting:
`EXEC sp_rename 'database_9fcf184fb06f4189b4e853dfc85b2b85.dbo."[test2]".column1', 'column3', 'COLUMN';`

So we can add support for renaming such stuff, even if don't support for simple select of such tables :)

Reply:
> But looks like we don't support to query such tables:
> `select * from "[test2]";`
> `uery 20220924_171606_00044_winm8 failed: Table 'dbo.[test2]' has no supported columns (all 0 columns are not supported)`

I checked that we cannot select tables with brackets in names because we didn't escape properly these table names:
because in `metadata getColumns` method, this parameter is used as SQL Like pattern and we should escape special characters


      `  return metadata.getColumns(`
               ` remoteTableName.getCatalogName().orElse(null),`
                `escapeNamePattern(remoteTableName.getSchemaName(), metadata.getSearchStringEscape()).orElse(null),`
                `escapeNamePattern(remoteTableName.getTableName(), metadata.getSearchStringEscape()),`
              `  null);`
`    }`

So I made a simple fix, by escaping brackets and now was I able to query such tables in trino:
`trino:dbo> SELECT * from sqlserver.dbo."[test3]";`
` column1 | column2`

Reply:
https://github.com/trinodb/trino/pull/14286